lms/services/file.py

```diff
@@ -9,30 +11,28 @@ def __init__(self, application_instance, db):
         self._application_instance = application_instance
         self._db = db
 
-    def get(self, lms_id, type_):
-        """Return the file with the given lms_id and type_ or None."""
-        return (
-            self._db.query(File)
-            .filter_by(
-                application_instance=self._application_instance,
-                lms_id=lms_id,
-                type=type_,
-            )
-            .one_or_none()
-        )
+    def get(self, lms_id, type_, course_id=None) -> Optional[File]:
+        """Return the file with the given parameters or None."""
+        return self._file_search_query(
+            application_instance=self._application_instance,
+            lms_id=lms_id,
+            type_=type_,
+            course_id=course_id,
+        ).one_or_none()
 
     def find_copied_file(self, new_course_id, original_file: File):
         """Find an equivalent file to `original_file` in our DB."""
         return (
-            self._db.query(File).filter(
+            self._file_search_query(
                 # Both files should be in the same application instance
-                File.application_instance == original_file.application_instance,
-                # Looking for files in the new course only
-                File.course_id == new_course_id,
+                application_instance=original_file.application_instance,
                 # Files of the same type
-                File.type == original_file.type,
+                type_=original_file.type,
+                # Looking for files in the new course only
+                course_id=new_course_id,
+            ).filter(
                 # We don't want to find the same file we are looking for
-                File.lms_id != original_file.lms_id,
```

Comment:
I reckon this comment was right, but the code was wrong.

To avoid finding the same file we exclude our PK in the table. Not the ID on the LMS which as we've seen can be duplicated (in different courses).